sdk/search/Azure.Search.Documents/src/Options/SearchOptions.cs

```diff
@@ -19,6 +19,77 @@ namespace Azure.Search.Documents
+        /// Please note <see cref="VectorQuery"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+        /// The available derived classes include <see cref="VectorizedQuery"/>.
+        /// </param>
+        /// <param name="filterMode"> Determines whether or not filters are applied before or after the vector search is performed. Default is 'preFilter' for new indexes. </param>
+        internal SearchOptions(bool? includeTotalCount, IList<string> facets, string filter, string highlightFieldsRaw, string highlightPostTag, string highlightPreTag, double? minimumCoverage, string orderByRaw, SearchQueryType? queryType, ScoringStatistics? scoringStatistics, string sessionId, IList<string> scoringParameters, string scoringProfile, string searchText, string searchFieldsRaw, SearchMode? searchMode, string selectRaw, int? skip, int? size, string semanticConfigurationName, SemanticErrorMode? semanticErrorMode, int? semanticMaxWaitInMilliseconds, string queryAnswerRaw, string queryCaptionRaw, IList<VectorQuery> vectorQueries, VectorFilterMode? filterMode)
+        {
+            IncludeTotalCount = includeTotalCount;
+            Facets = facets;
+            Filter = filter;
+            HighlightFieldsRaw = highlightFieldsRaw;
+            HighlightPostTag = highlightPostTag;
+            HighlightPreTag = highlightPreTag;
+            MinimumCoverage = minimumCoverage;
+            OrderByRaw = orderByRaw;
+            QueryType = queryType;
+            ScoringStatistics = scoringStatistics;
+            SessionId = sessionId;
+            ScoringParameters = scoringParameters;
+            ScoringProfile = scoringProfile;
+            SearchText = searchText;
+            SearchFieldsRaw = searchFieldsRaw;
+            SearchMode = searchMode;
+            SelectRaw = selectRaw;
+            Skip = skip;
+            Size = size;
+
+            SemanticSearch = (semanticConfigurationName != null || semanticErrorMode != null || semanticMaxWaitInMilliseconds != null || queryAnswerRaw != null || queryCaptionRaw != null) ? new SemanticSearchOptions() : null;
+            SemanticConfigurationName = semanticConfigurationName;
+            SemanticErrorMode = semanticErrorMode;
+            SemanticMaxWaitInMilliseconds = semanticMaxWaitInMilliseconds;
+            if (SemanticSearch != null)
+            {
+                SemanticSearch.QueryAnswer = queryAnswerRaw != null ? new QueryAnswer() : null;
+                QueryAnswerRaw = queryAnswerRaw;
+
+                SemanticSearch.QueryCaption = queryCaptionRaw != null ? new QueryCaption() : null;
+                QueryCaptionRaw = queryCaptionRaw;
+            }
+
+            VectorSearch = (vectorQueries != null || filterMode != null) ? new VectorSearchOptions() : null;
+            VectorQueries = vectorQueries;
+            FilterMode = filterMode;
```

Comment:
The issue occurred because the generator didn't know about the new types—`SemanticSearch`, `SemanticSearch.QueryAnswer`, `SemanticSearch.QueryCaption`, and `VectorSearch`. As a result, it didn't initialize these types to use the same search options to call the next page request. The fix is to initialize these types if any semantic or vector search properties are present.